astropy/units/tests/test_quantity_non_ufuncs.py

```diff
@@ -2578,3 +2585,170 @@ def test_all_included(self):
+    We explicitly allow helper's signature to be *more* flexible than the target signature
+    by allowing *args and **kwargs catch-all arguments, which we use to limit code
+    duplication, and also help with forward and backward compatibility.
+    See https://github.com/astropy/astropy/issues/15703
+    """
+
+    @staticmethod
+    def _have_catchall_argument(parameters, kind) -> bool:
+        return any(p.kind is kind for p in parameters.values())
+
+    @staticmethod
+    def _get_param_group(parameters, kinds: list) -> list[str]:
+        return [name for name, p in parameters.items() if p.kind in kinds]
+
+    def test_all_arguments_reexposed(self, functions):
+        target, helper = functions
+
+        try:
+            sig_target = inspect.signature(target)
+        except ValueError:
+            pytest.skip("Non Python function cannot be inspected at runtime")
+
+        params_target = sig_target.parameters
+        sig_helper = inspect.signature(helper)
+        params_helper = sig_helper.parameters
+
+        have_args_helper = self._have_catchall_argument(params_helper, VAR_POSITIONAL)
+        have_kwargs_helper = self._have_catchall_argument(params_helper, VAR_KEYWORD)
+
+        args_target = list(params_target.items())
+        args_helper = list(params_helper.items())
+
+        pos_target = 0
+        pos_helper = 0
+        while pos_target < len(args_target):
+            nt, pt = args_target[pos_target]
+            kt = pt.kind
+            if kt in (POSITIONAL_ONLY, POSITIONAL_OR_KEYWORD):
+                assert pos_helper < len(args_helper), (
+                    "helper's signature is too short; "
+                    "some arguments are not properly re-exposed"
+                )
+                nh, ph = args_helper[pos_helper]
+                if (kh := ph.kind) is not VAR_POSITIONAL:
+                    assert nh == nt, f"argument {nt!r} isn't re-exposed as positional"
+                    assert kh is kt, (
+                        f"helper is not re-exposing argument {nt!r} properly:"
+                        f"expected {kt}, got {kh}"
+                    )
+                    pos_helper += 1
+            if kt in (KEYWORD_ONLY, POSITIONAL_OR_KEYWORD):
+                if nt in params_helper:
+                    assert (kh := params_helper[nt].kind) is kt, (
+                        f"helper is not re-exposing argument {nt!r} properly: "
+                        f"expected {kt}, got {kh}"
+                    )
+                elif kt is KEYWORD_ONLY:
+                    assert (
+                        have_kwargs_helper
+                    ), f"argument {nt!r} is not re-exposed as keyword"
+                elif kt is POSITIONAL_OR_KEYWORD:
+                    assert (
+                        have_args_helper and have_kwargs_helper
+                    ), f"argument {nt!r} is not re-exposed as positional-or-keyword"
+            if pt.kind is VAR_POSITIONAL:
```

Comment:
replace `pt.kind` with `kt` here too (or use `pt.kind` everywhere -- may be a bit clearer)

Reply:
Also, this can be an `elif`

Reply:
`elif` statements seem redundant with `continue` (which I just added following another of your suggestions !), and having them all be `if` makes it slightly more readable (to me ?) because it forces alignement and makes it easier to spot differences.